sdk/metric/config.go

```diff
@@ -123,10 +123,7 @@ func WithResource(res *resource.Resource) Option {
 // operations; no data will be exported without a Reader.
 func WithReader(r Reader, views ...view.View) Option {
 	return optionFunc(func(cfg config) config {
-		if cfg.readers == nil {
-			cfg.readers = make(map[Reader][]view.View)
-		}
-		cfg.readers[r] = views
+		cfg.viewers = append(cfg.viewers, viewer{reader: r, views: views})
```

Comment:
This seems to admit having multiple `viewer` containing the same `Reader`, which will result in `Shutdown()` and `ForceFlush()` being called on that `Reader` multiple times.  Is that safe?  It doesn't look like any of the test cases cover that situation.

Reply:
Calling these methods multiple times is indeed safe for our readers [^1][^2]. Furthermore, there will not be a duplicate call to collection because of their guards to not be registered multiple times.

[^1]: https://github.com/open-telemetry/opentelemetry-go/blob/b6a22abab751307b4851b3cc51728fa557f907bb/sdk/metric/manual_reader.go#L71-L87
[^2]: https://github.com/open-telemetry/opentelemetry-go/blob/b6a22abab751307b4851b3cc51728fa557f907bb/sdk/metric/periodic_reader.go#L242-L292

Reply:
I can look into adding tests check what happens when registering these two types readers multiple times, but I'm not sure what we want to check in those situations. Were you just wanting to make sure they don't panic?